Modules/CPackDeb.cmake

```diff
@@ -630,6 +694,22 @@ function(cpack_deb_prepare_package_vars)
         endif()
       endif()
     endforeach()
+    set(COMPONENT_DEPENDS "")
+    foreach (_PACK ${CPACK_COMPONENT_${_local_component_name}_DEPENDS})
+      get_component_package_name(_PACK_NAME "${_PACK}")
+      if(COMPONENT_DEPENDS)
+        set(COMPONENT_DEPENDS "${_PACK_NAME} (= ${CPACK_DEBIAN_PACKAGE_VERSION}), ${COMPONENT_DEPENDS}")
+      else()
+        set(COMPONENT_DEPENDS "${_PACK_NAME} (= ${CPACK_DEBIAN_PACKAGE_VERSION})")
+      endif()
+    endforeach()
+    if(COMPONENT_DEPENDS)
+      if(CPACK_DEBIAN_PACKAGE_DEPENDS)
+        set(CPACK_DEBIAN_PACKAGE_DEPENDS "${COMPONENT_DEPENDS}, ${CPACK_DEBIAN_PACKAGE_DEPENDS}")
```

Comment:
Adding automatic dependencies between components is also an incompatible change.